[a11y] misc admin modules

brianteeman opened the pull request:
Continuing the work to prevent assistive technology reading out the value of an icon. This PR addresses the icons as shown below

logout button on mod_logged
<img width="385" alt="screenshotr11-41-38" src="https://cloud.githubusercontent.com/assets/1296369/24604151/95dadca6-185b-11e7-9f8d-67658f3a8358.png">

multilingual status icon
<img width="320" alt="screenshotr11-43-58" src="https://cloud.githubusercontent.com/assets/1296369/24604152/95dae336-185b-11e7-8d49-f00c036bf8ae.png">

admin stats modules line icons
<img width="402" alt="screenshotr11-45-23" src="https://cloud.githubusercontent.com/assets/1296369/24604153/95dfde9a-185b-11e7-8ad2-87df47fcff6c.png">

admin status module
<img width="456" alt="screenshotr11-48-23" src="https://cloud.githubusercontent.com/assets/1296369/24604154/95e3e4ae-185b-11e7-859e-ab60c23a4e0b.png">

home icon on default menu
<img width="270" alt="screenshotr11-50-12" src="https://cloud.githubusercontent.com/assets/1296369/24604155/95e96406-185b-11e7-8438-e37a03b9ef59.png">

C-Lodder commented on administrator/modules/mod_logged/tmpl/default.php:
```diff
@@ -17,7 +17,7 @@
 				<?php if ($user->client_id == 0) : ?>
 					<a title="<?php echo JHtml::_('tooltipText', 'MOD_LOGGED_LOGOUT'); ?>" href="<?php echo $user->logoutLink; ?>" class="btn btn-danger btn-mini hasTooltip">
-						<span class="icon-remove icon-white" title="<?php echo JText::_('JLOGOUT'); ?>"></span>
+						<span class="icon-remove icon-white" aria-hidden="true" title="<?php echo JText::_('JLOGOUT'); ?>"></span>
```
The `JText::_('JLOGOUT');` need to go into a separate `span`. Currently, nothing will be provided on screen readers

C-Lodder replied:
This doesn't look right. You're hiding the icon on screen readers, so there will be an empty `<a>` element

brianteeman replied:
no there is a title in the a link - at least thats my understanding of it

C-Lodder replied:
hmm, cause if inline elements have no content inside them, they end up being `0px` in width and height, therefore you can't hover over them. That would be the case here, in which case the title won't be read out.

brianteeman replied:
In supporting browsers in conjunction with supporting assistive technology the content is not conveyed to the user via the assistive technology.
Content is displayed in the browser.

Source https://www.paciellogroup.com/blog/2012/05/html5-accessibility-chops-hidden-and-aria-hidden/

Quy replied:
Related #14981. 

From fontawesome.io:
>In the case of focusable interactive elements, there are various options to include an alternative text or label to the element, without the need for any visually hidden <span> or similar. For instance, simply adding the aria-label attribute with a text description to the interactive element itself will be sufficient to provide an accessible alternative name for the element. If you need to provide a visual tooltip on mouseover/focus, we recommend additionally using the title attribute or a custom tooltip solution.
```
<a class="btn btn-danger" href="path/to/settings" aria-label="Delete">
  <i class="fa fa-trash-o" aria-hidden="true" title="Delete this item?"></i>
</a>
```
Read this page for more information; http://fontawesome.io/accessibility/

tonypartridge replied:
Do we not need to include aria-label @brianteeman as per @Quy since this is how I read the FA Accessibility FAQ's.

brianteeman replied:
I will double check this one

brianteeman committed:
logout button on mod_logged

brianteeman committed:
multilingual status icon

brianteeman committed:
admin stats modules line icons

brianteeman committed:
admin status module

brianteeman committed:
home icon on default menu

brianteeman committed:
Merge branch 'staging' into a11y_admin_modules

brianteeman committed:
update

I went for the `<span class="element-invisible">` instead of the aria-title so it is more consistent with other uses elsewhere

brianteeman commented:
thanks for testing - I made a requested change so it needs retesting - thanks in advance

Quy commented:
I have tested this item :white_check_mark: successfully on e969192a65ca2f0afcb26b157f0de41a719b6465<br /><br />Code review.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15068">issues.joomla.org/tracker/joomla-cms/15068</a>.</sub>

franz-wohlkoenig commented:
I have tested this item :white_check_mark: successfully on e969192a65ca2f0afcb26b157f0de41a719b6465<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15068">issues.joomla.org/tracker/joomla-cms/15068</a>.</sub>

franz-wohlkoenig commented:
RTC after two successful tests.

rdeutz committed:
Merge branch 'staging' into a11y_admin_modules

brianteeman commented:
Thanks - one more small step to improving a11y in joomla
